reana_client/api/client.py

```diff
@@ -420,3 +430,96 @@ def get_analysis_code(self, user, organization, analysis_id):
+                raise FileUploadError('Path cannot contain ".."')
+
+            # Check if input is a directory and upload everything
+            # including subdirectories.
+            if os.path.isdir(path):
+                logging.debug("'{}' is a directory.".format(path))
+                logging.info("Uploading contents of folder '{}' ..."
+                             .format(path))
+                for root, dirs, files in os.walk(path, topdown=False):
+                    responses = []
+                    for next_path in files + dirs:
+                        try:
+                            response = self.upload_to_server(
+                                user, organization, workflow,
+                                os.path.join(root, next_path),
+                                upload_type)
+                            responses.append(os.path.join(root, next_path))
+                        except Exception as e:
+                            logging.error(traceback.format_exc())
+                            logging.error("Something went wrong"
+                                          " while uploading '{0}'".
+                                          format(os.path.join(root,
+                                                              next_path)))
+                return responses
+
+            # Check if input is an absolute path and upload file.
+            else:
+                with open(path) as f:
+                    fname = os.path.basename(f.name)
+                    if len(path.split('/')) > 1 and \
+                            path.split('/')[0] == UploadType(upload_type).name:
+                        path = "/".join(path.strip("/").split('/')[1:])
+                    logging.debug("'{}' is an absolute filepath."
+                                  .format(os.path.basename(fname)))
+                    logging.info("Uploading '{}' ...".format(fname))
+                    try:
+                        if upload_type is UploadType.code:
+                            response = self.seed_analysis_code(
+                                user, organization, workflow, f, path)
+                        elif upload_type is UploadType.inputs:
+                            response = self.seed_analysis_inputs(
+                                user, organization, workflow, f, path)
+                        else:
+                            logging.warning("Unknown upload type of '{}'."
+                                            "File '{}' was not uploaded."
+                                            .format(upload_type, path))
+                        if response:
+                            logging.info("File '{}' was successfully "
+                                         "uploaded.".format(fname))
+                        return response
+                    except Exception as e:
+                        logging.error(traceback.format_exc())
+                        logging.error("Something went wrong"
```

Comment:
Currently this gives:

```console
$ reana-client code upload ./code/somethingthatdoesnotexist
Something went wrong while uploading ./code/somethingthatdoesnotexist
```

This looks obscure; it would be nicer to say that the file or directory does not exist.

Can you please introduce a check whether the input is a file or directory and fail with an appropriate error message? E.g. does not exist, does not have read permissions, etc...

Reply:
Could we return the exception message?
For a file or dir that doesn't exist it would show:
```
(reana-cwl) ➜  reana-demo-root6-roofit git:(master) ✗ reana-client code upload code/code_3
[Errno 2] No such file or directory: 'code/code_3'
```

I could add the check for the 2 cases, but this would show information about other exceptions that could occur.

Reply:
Right, in the last changes, click.Path(exists=True, ...) shows a nice message, and the exception message is reverted.